test/spear_test.exs

```diff
@@ -978,6 +1005,48 @@ defmodule SpearTest do
       assert Spear.stream!(c.conn, c.stream_name) |> Enum.map(& &1.body) == Enum.to_list(0..19)
     end
 
+    @tag compatible(">= 21.6.0")
+    test "the append_batch/5 with `raw?: false` returns :ok", c do
+      assert {:ok, batch_id, request_id} =
+               random_events()
+               |> Stream.take(5)
+               |> Spear.append_batch(c.conn, :new, c.stream_name, expect: :empty, raw?: false)
+
+      assert_receive %Spear.BatchAppendResult{
+        result: result,
+        batch_id: ^batch_id,
+        request_id: ^request_id,
+        revision: 4
+      }
+
+      assert :ok = result
+
+      assert Spear.cancel_subscription(c.conn, request_id) == :ok
+
+      assert Spear.stream!(c.conn, c.stream_name) |> Enum.map(& &1.body) == Enum.to_list(0..4)
+    end
+
+    @tag compatible(">= 21.6.0")
+    test "the append_batch/5 with `raw?: true` returns the raw result", c do
+      assert {:ok, batch_id, request_id} =
+               random_events()
+               |> Stream.take(5)
+               |> Spear.append_batch(c.conn, :new, c.stream_name, expect: :empty, raw?: true)
+
+      assert_receive %Spear.BatchAppendResult{
+        result: result,
+        batch_id: ^batch_id,
+        request_id: ^request_id,
+        revision: 4
+      }
+
+      assert {:success, Streams.batch_append_resp_success()} = result
```

Comment:
It's weird that this behaves differently on Elixir 1.7. I want to bump the versions anyway though so I will merge this as-is